### PR TITLE
Edit gamecard

### PR DESCRIPTION
**Pull request by JanaHaeusler**
The user can now edit the respective gamecard. By clicking on the edit button the form opens in an edit modus and shows the content of the selected gamecard, which can be updated. I implemented a useForm hook (thereby downsized the form component) and cleaned the code. Additionally the button structure and icon imports are updated. 
I am sorry that this PR got larger than usual by updating and cleaning up. Next time I'll split it in a better way. :) 
The most important things are the useForm hook and the editing functionality (GameCard.js, GameCardsList.js, SaveGameForm.js, App.js, useForm.js, useGameData.js). I really appreciate any thoughts and comments. Thank you very, very much!

### CODE REVIEW

File: src/components/GameCard.js [hunk 1]
```diff
@@ -75,58 +78,69 @@ const Location = styled.div`
     text-transform: uppercase;
 `
 const PlayerWrapper = styled.div`
-    grid-column: 1/4;
+    grid-column: 1 / 4;
     grid-row-start: 3;
 `
 const WinnerWrapper = styled.div`
-    grid-column: 1;
+    grid-column-start: 1;
```
Comment (CinFrei) [inline src/components/GameCard.js:85]: Css hacker 👍

**Review by CinFrei — COMMENTED** [review]
Good Work 🚀

**Review by ncbecker — COMMENTED** [review]
Nice work! And great to see testing implemented 🚀🚀

File: src/components/Buttons/ButtonPrimary.js [hunk 2]
```diff
@@ -19,7 +22,7 @@ const ButtonStyled = styled.button`
 const ButtonPrimary = (props) => <ButtonStyled {...props} />
 
 ButtonPrimary.propTypes = {
-    children: PropTypes.string.isRequired,
+    children: PropTypes.array.isRequired,
     disabled: PropTypes.bool,
     onClick: PropTypes.func,
```
Comment (Hecktors) [inline src/components/Buttons/ButtonPrimary.js:27]: Don't you always expect an `onClick` function? Maybe you might as well set it to` .isRequired`...

File: src/components/Buttons/ButtonPrimary.js [hunk 3]
```diff
@@ -19,7 +22,7 @@ const ButtonStyled = styled.button`
 const ButtonPrimary = (props) => <ButtonStyled {...props} />
```
Comment (Hecktors) [inline src/components/Buttons/ButtonPrimary.js:22]: Love how you handle your styled depending attribute `disabled`. I'll rob it!

File: src/components/GameCard.js [hunk 4]
```diff
@@ -26,25 +27,28 @@ export default function GameCard({location, date, players, winner, shots, id, on
...
             {isSetToDelete && (
                 <DeleteField>
                     <span>Do you want to delete this gamecard?</span>
-                    <ButtonDeleteSecond onClick={() => onDelete(id)}><BinIcon/>Delete</ButtonDeleteSecond>
-                    <ButtonCancel onClick={() => setIsSetToDelete(false)}> X Cancel</ButtonCancel>
+                    <ButtonDelete onClick={() => onDelete(id)} data-testid="button-delete"><BinIcon/>Delete</ButtonDelete>
+                    <ButtonCancel onClick={() => setIsSetToDelete(false)} data-testid="button-cancel"><CancelIcon/>Cancel</ButtonCancel>
                 </DeleteField>
             )}
         </Card>
```
Comment (Hecktors) [inline src/components/GameCard.js:54]: Great UX. So nice that logic!

File: src/components/Icons.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import { ReactComponent as Check } from '../assets/check.svg'
+import { ReactComponent as Cross } from '../assets/cross.svg'
+import { ReactComponent as Bin } from '../assets/bin.svg'
+import { ReactComponent as Pencil } from '../assets/pencil.svg'
+
+export { Check, Cross, Bin, Pencil }
```
Comment (Hecktors) [inline src/components/Icons.js:6]: Such a great idea to keep the code clean!   😍

File: src/App.js [hunk 6]
```diff
@@ -5,19 +5,47 @@ import GameCardsList from './components/GameCardsList'
...
 const AppWrapper = styled.div`
+  padding: 10px;
+  margin: 5px auto;
+  max-width: 600px;
+`
+const SaveGameFormWrapper = styled.div`
   display: grid;
   grid-gap: 20px;
-`
+`
```
Comment (Hecktors) [inline src/App.js:51]: Awesome clean App.js 👍

File: src/hooks/useForm.js [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+            date: '',
+            players: '',
+            winner: '',
+            shots: '',
+            _id: '',
+        } 
+        saveLocally(STORAGE_KEY, emptyInputs)
+        cancelEditModus(false)
+    }
+}
```
Comment (Hecktors) [inline src/hooks/useForm.js:155]: Super awesome

**Review by Hecktors — COMMENTED** [review]
Beautiful code and structure! Thumb up 👍👏

**Pull request merged**